src/contracts/core/AllocationManager.sol

```diff
@@ -645,6 +645,71 @@ contract AllocationManager is
         return totalMagnitude;
     }
 
+    /**
+     * @notice Returns the pending allocations of an operator for a given strategy and operatorSets
+     * @param operator the operator to get the pending allocations for
+     * @param strategy the strategy to get the pending allocations for
+     * @param operatorSets the operatorSets to get the pending allocations for
+     * @return pendingMagnitude the pending allocations for each operatorSet
+     * @return timestamps the timestamps for each pending allocation
+     */
+    function getPendingAllocations(
+        address operator,
+        IStrategy strategy,
+        OperatorSet[] calldata operatorSets
+    ) external view returns (uint64[] memory, uint64[] memory) {
+        uint64[] memory pendingMagnitude = new uint64[](operatorSets.length);
+        uint64[] memory timestamps = new uint64[](operatorSets.length);
+        for (uint256 i = 0; i < operatorSets.length; ++i) {
+            (bool exists, uint32 key, uint224 value) =
+                _magnitudeUpdate[operator][strategy][_encodeOperatorSet(operatorSets[i])].latestSnapshot();
+            if (exists) {
+                if (key > block.timestamp) {
+                    pendingMagnitude[i] = uint64(value);
+                    timestamps[i] = key;
+                } else {
+                    pendingMagnitude[i] = 0;
+                    timestamps[i] = 0;
+                }
+            }
+        }
+        return (pendingMagnitude, timestamps);
```

Comment:
how come you convert the timestamps from uint32 -> uint64?

Reply:
oh I actually did not realize that - good catch. `forge b` did not detect it :O

Reply:
It's because it casted the uint and didn't perceive that as an error

Reply:
ahh auto casting. will change it to uint32 - it was not my intention. good catch